tests/tests/mempool.rs

```diff
@@ -0,0 +1,396 @@
+        .await
+        .expect("Failed to create sample transaction")
+        .pop()
+        .expect("Expected at least one transaction");
+
+    // Check initial pool size
+    assert_eq!(eth_provider.mempool().unwrap().pool_size().total, 0);
+
+    // Add transaction to mempool
+    let result = eth_provider.mempool().unwrap().add_transaction(TransactionOrigin::Local, transaction.clone()).await;
+
+    // Ensure the transaction was added successfully
+    assert!(result.is_ok());
+
+    // Get updated mempool size
+    let mempool_size = eth_provider.mempool().unwrap().pool_size();
+
+    // Get the EOA address
+    let address = katana.eoa().evm_address().expect("Failed to get eoa address");
+
+    // Get transactions by sender address and nonce
+    let sender_transactions = eth_provider.mempool().unwrap().get_transactions_by_sender_and_nonce(address, 0);
+
+    // Get transactions by origin
+    let origin_transaction = eth_provider.mempool().unwrap().get_transactions_by_origin(TransactionOrigin::Local);
+
+    // Get local transactions
+    let local_transaction = eth_provider.mempool().unwrap().get_local_transactions();
+
+    // Get all transactions in the mempool
+    let all_transactions = eth_provider.mempool().unwrap().all_transactions();
+
+    // Check pending, queued and total transactions
+    assert_eq!(mempool_size.pending, 1);
+    assert_eq!(mempool_size.queued, 0);
+    assert_eq!(mempool_size.total, 1);
+
+    // get_transactions_by_sender_and_nonce test
+    // Check if the returned transaction hash matches
+    assert_eq!(*sender_transactions.unwrap().hash(), transaction_signed.hash());
+
+    // get_transactions_by_origin function test
+    // Check if the returned transaction hash matches
+    assert_eq!(*origin_transaction[0].hash(), transaction_signed.hash());
+
+    // get_local_transactions function test
+    // Check if the returned transaction hash matches
+    assert_eq!(*local_transaction[0].hash(), transaction_signed.hash());
+    assert_eq!(*local_transaction[0].hash(), *origin_transaction[0].hash());
+
+    // Remove transaction by hash
+    let _ = eth_provider.mempool().unwrap().remove_transactions(vec![transaction_signed.hash()]);
+
+    // remove_transactions function tests
+    // Get updated mempool size
+    let mempool_size = eth_provider.mempool().unwrap().pool_size();
+    // Check pending, queued and total transactions after remove_transactions
+    assert_eq!(mempool_size.pending, 0);
+    assert_eq!(mempool_size.queued, 0);
+    assert_eq!(mempool_size.total, 0);
+
+    // all_transactions function tests
+    // Check if the first pending transaction hash matches
+    assert_eq!(*all_transactions.pending[0].hash(), transaction_signed.hash());
+    // Ensure only one pending transaction is present
+    assert_eq!(all_transactions.pending.len(), 1);
+    // Ensure no queued transactions are present
+    assert_eq!(all_transactions.queued.len(), 0);
```

Comment:
can we move this above the `remove_transaction` test? It's a little confusing when reading the test